packages/zipkin-transport-http/test-types/HttpLogger.test.ts

```diff
@@ -0,0 +1,37 @@
+import { expect } from 'chai';
+
+import {Agent as HttpAgent } from 'http';
+import {Agent as HttpsAgent} from 'https';
+import {jsonEncoder} from 'zipkin';
+
+import { HttpLogger } from 'zipkin-transport-http';
+
+describe('HttpLogger', () => {
+  it('should have correct type', () => {
+    const options = {
+      endpoint: 'testEndpoint',
+      httpInterval: 1000,
+      jsonEncoder: jsonEncoder.JSON_V1,
+      timeout: 0,
+      maxPayloadSize: 0,
+      headers: {},
+      agent: new HttpAgent(),
+      log: console
+    };
+    const httpLogger: HttpLogger = new HttpLogger(options);
+
+    expect(httpLogger.logSpan).to.be.a('function');
+  });
+
+  it('should accept Http(s) Agent or function which returns Agent', () => {
+    const agents = [new HttpAgent(), new HttpsAgent(), () => new HttpAgent(), () => new HttpsAgent()];
```

Comment:
so this test makes sure it doesn't crash in the constructor. am I understanding that right?

I wonder if it would crash later. It might be confusing to have a test without assertions I mean.. wdyt?

Reply:
Ya it's testing the constructor. I've added an expect statement to confirm `this.agent` is properly being set and defaulting to `null` if it is not provided.

Beyond that, it is only being forwarded to the fetch call. Ideally I could confirm that fetch is being called with the agent, however due to the `node-fetch` implementation and `sinon` restrictions, it doesn't seem feasible to perform such a test without some hacky solutions.